lib/rate-limit.js

```diff
@@ -1,39 +1,47 @@
-export default function rateLimit (instance, maxRetry = 5) {
-  let attempt = 0
-  let lastErrorAt
+const attempts = {}
 
+export default function rateLimit (instance, maxRetry = 5) {
   instance.interceptors.response.use(function (response) {
     // we don't need to do anything here
     return response
   }, function (error) {
-    const {config} = error
-    if (!config || !instance.defaults.retryOnError) {
+    const {request, response, config} = error
+    if ((!request && !response) || !config || !instance.defaults.retryOnError) {
       return Promise.reject(error)
     }
-    // in axios reponse will be available only if there a response from server
-    // in this case will assume it is a 500 error
-    error.response = error.response || {status: 500}
-    if (error.response.status === 500) {
-      attempt++
-      lastErrorAt = lastErrorAt || Date.now()
-      lastErrorAt = Date.now()
-      // we reject if there are too much errors in a short period of time
-      if (attempt >= maxRetry) {
+
+    let retryErrorType = null
+    let wait = 0
+
+    if (!response) {
+      // Errors without response or config did not recieve anything from the server
+      retryErrorType = 'Connection'
+    } else if (response.status === 500) {
```

Comment:
`5XX` errors (not just 500) should all be retried as our backend uses them to signal a temporary failure, which should resolve eventually.